posthog/api/dashboard.py

```diff
@@ -26,6 +26,15 @@
 from posthog.utils import render_template
 
 
+class CanEditDashboard(BasePermission):
```

Comment:
As a side note, feels like our permissions engine is getting quite convoluted, and will get harder to maintain. Previously I've used [DRY permissions](https://github.com/dbkaplan/dry-rest-permissions), I believe it's no longer maintained but the model was quite useful, you would add `has_object_permission` and/or `has_permission` methods to all models, and views would pull from there. That way you have a central place to understand permissions.